server/services/offer.service.js

```diff
@@ -270,6 +271,22 @@ export const getOffer = async (offerId, user) => {
   return offer;
 };
 
+const generatePaymentDates = (offer) => {
+  const paymentDates = [];
+  const { totalAmountPayable, initialPayment, periodicPayment, paymentFrequency } = offer;
+
+  const numberOfPaymentsToBeMade = (totalAmountPayable - initialPayment) / periodicPayment;
+
+  for (let i = 0; i < numberOfPaymentsToBeMade; i += 1) {
+    const handOverDate =
+      offer.handOverDate === 'string' ? parseISO(offer.handOverDate) : offer.handOverDate;
```

Comment:
I think it is better to remove this since we already know the cause of the failing tests. It is an hot fix, so we should only use it if it is becoming difficult to troubleshoot the root cause.

Also, this should be 
```suggestion
      typeof offer.handOverDate === 'string' ? parseISO(offer.handOverDate) : offer.handOverDate;
```